role.module

```diff
@@ -24,39 +25,36 @@ function role_entity_type_alter(array &$entity_types) {
  * Implements hook_entity_form_display_alter().
  */
 function role_entity_form_display_alter(EntityFormDisplayInterface &$form_display, array $context) {
-  if ($context['entity_type'] == 'user' && $context['bundle'] == 'user') {
+  if ($context['entity_type'] === 'user' && $context['bundle'] === 'user') {
+    /** @var Drupal\role\Service\RoleControlManagerInterface $control_manager */
     $control_manager = Drupal::service('role.control_manager');
     /** @var \Drupal\user\Entity\User $user */
     $user = Drupal::routeMatch()->getParameter('user');
     if (!$user) {
       return;
     }
 
-    /** @var Drupal\user\Entity\Role $role */
+    /** @var Drupal\user\Entity\RoleInterface $role */
     $role = $control_manager->getUserPriorityRole($user);
 
     if ($role) {
-      // Set form display mode to the edit account page.
-      $third_party_settings = $role->getThirdPartySettings('role');
-      if ($third_party_settings) {
-        /** @var \Drupal\Core\Config\Entity\ConfigEntityStorageInterface $storage */
-        $storage = Drupal::service('entity_type.manager')
-          ->getStorage('entity_form_display');
-
-        if (!isset($third_party_settings['account_form_mode'])) {
-          $authenticated_role = Role::load('authenticated');
-          $default_form_mode = $authenticated_role->getThirdPartySettings('role')['account_form_mode'];
-          if (!$default_form_mode) {
-            $default_form_mode = 'default';
-          }
-          $form_display = $storage->load($context['entity_type'] . '.' . $context['bundle'] . '.' . $default_form_mode);
+      /** @var \Drupal\Core\Config\Entity\ConfigEntityStorageInterface $storage */
+      $storage = Drupal::service('entity_type.manager')
+        ->getStorage('entity_form_display');
+
+      if (!$control_manager->getUserAccountFormMode($user)) {
+        $authenticated_role = Role::load(AccountInterface::AUTHENTICATED_ROLE);
```

Comment:
These code is unnecessary, because your method `getUserAccountFormMode` already does it.